drift-codec/src/main/java/com/facebook/drift/codec/internal/builtin/UriThriftCodec.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2012 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.drift.codec.internal.builtin;
+
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.codec.metadata.ThriftType;
+import com.facebook.drift.protocol.TProtocolReader;
+import com.facebook.drift.protocol.TProtocolWriter;
+
+import java.net.URI;
+
+import static java.util.Objects.requireNonNull;
+
+public class UriThriftCodec
```

Comment:
I don't think you need to add a new default type.  Try this and see if it works: https://gist.github.com/tdcmeehan/3b53ee87fdd11c69fd81559df0f0ee17

Reply:
done. please review when you get a chance.